app/src/androidTest/java/com/duckduckgo/app/notification/NotificationSchedulerTest.kt

```diff
@@ -52,24 +55,47 @@ class NotificationSchedulerTest {
     }
 
     @Test
-    fun whenClearNotificationCanShowThenNotificationScheduled() = runBlocking<Unit> {
+    fun whenBothPrivacyNotificationAndCleatDataCanShowThenPrivacyNotificationScheduled() = runBlocking<Unit> {
+        whenever(privacyNotification.canShow()).thenReturn(true)
         whenever(clearNotification.canShow()).thenReturn(true)
         testee.scheduleNextNotification()
-        assertTrue(notificationScheduled())
+        assertTrue(notificationScheduled(PrivacyNotificationWorker::class.jvmName))
     }
 
     @Test
-    fun whenClearNotificationCannotShowThenNotificationNotScheduled() = runBlocking<Unit> {
+    fun whenPrivacyNotificationCanShowAndCleatDataCannotThenPrivacyNotificationScheduled() = runBlocking<Unit> {
+        whenever(privacyNotification.canShow()).thenReturn(true)
+        whenever(clearNotification.canShow()).thenReturn(false)
+        testee.scheduleNextNotification()
+        assertTrue(notificationScheduled(PrivacyNotificationWorker::class.jvmName))
+    }
+
+    @Test
+    fun whenPrivacyNotificationCannotShowAndClearNotificationCanShowThenNotificationScheduled() = runBlocking<Unit> {
+        whenever(privacyNotification.canShow()).thenReturn(false)
+        whenever(clearNotification.canShow()).thenReturn(true)
+        testee.scheduleNextNotification()
+        assertTrue(notificationScheduled(ClearDataNotificationWorker::class.jvmName))
+    }
+
+    @Test
+    fun whenNoNotificationCanShowThenNoNotificationScheduled() = runBlocking<Unit> {
+        whenever(privacyNotification.canShow()).thenReturn(false)
         whenever(clearNotification.canShow()).thenReturn(false)
         testee.scheduleNextNotification()
         assertFalse(notificationScheduled())
     }
 
-    private fun notificationScheduled(): Boolean {
-        return WorkManager
+    private fun notificationScheduled(workerName: String? = null): Boolean {
```

Comment:
it feels odd that this parameter is `nullable`, and it seems like the only reason it can be is to allow `assertFalse(notificationScheduled())` to be executed.

I think it would make it more readable to make another function to assert that no notifications have been scheduled, and change the existing function to accept a non-nullable class name.

Could you replace `assertFalse(notificationScheduled())` with something like this...? `assertNoNotificationsScheduled()`

        private fun assertNoNotificationsScheduled() {
        val workers = WorkManager
            .getInstance()
            .getWorkInfosByTag(NotificationScheduler.WORK_REQUEST_TAG)
            .get()
            .filter{ it.state == WorkInfo.State.ENQUEUED }

        assertTrue(workers.isEmpty())
    }